app/models/search_facet.rb

```diff
@@ -267,9 +268,11 @@ def self.get_table_schema(table_name: CellMetadatum::BIGQUERY_TABLE, column_name
 
   # update all search facet filters after BQ update
   def self.update_all_facet_filters
```

Comment:
What's the performance/reliability of fetching the facets from Azul?  This means we have to make a query to azul every time a metadata file is updated, which seems like more than we need, and will make some tests more brittle.  Wondering if it's worth adding a `field :cached_azul_facets, type: {}` and then adding a parameter to this method (defaulting to false) for whether to refresh the cached azul facets too

Reply:
The performance for that particular query is very good - around 1s.  We're only asking for a single project, so the response time is quick.  As to the reliability, this might be an instance where the health check is a good thing, and if there's any kind of issue/error, we skip updating the filters from Azul.

Those values don't change super often, but the Azul team said it is possible for those values to update at least on a weekly basic, though bi-weekly is more normal.  That being said, we don't get a huge number of metadata files daily.  According to [Mixpanel](https://mixpanel.com/project/2120588/view/19059/app/insights#report/9671053/file-ingests-scp/~(columnWidths~(bar~())~description~'Number*20of*20ingests*20of*20study*20files*20--*20gene*20expression*20matrices*2c*20cluster*20files*2c*20and*20metadata*20files*20--*20in*20Single*20Cell*20Portal*2c*20done*20via*20Ingest*20Pipeline.~displayOptions~(chartType~'line~plotStyle~'stacked~analysis~'linear~value~'absolute)~sorting~(bar~(sortBy~'column~colSortAttrs~(~(sortBy~'value~sortOrder~'desc)~(sortBy~'value~sortOrder~'desc)))~line~(sortBy~'value~sortOrder~'desc~valueField~'averageValue~colSortAttrs~(~(sortBy~'value~sortOrder~'desc~valueField~'averageValue)))~table~(sortBy~'column~colSortAttrs~(~(sortBy~'label~sortOrder~'asc)~(sortBy~'label~sortOrder~'asc)))~insights-metric~(sortBy~'value~sortOrder~'desc~valueField~'totalValue~colSortAttrs~(~))~pie~(sortBy~'value~sortOrder~'desc~valueField~'totalValue~colSortAttrs~(~)))~timeComparison~null~liftComparison~null~querySamplingEnabled~false~dashboard~(id~647255~title~'Single*20Cell*20Portal*20over*20time~creator_name~'Eric*20Weitz~can_update_basic~true~can_share~true~allow_staff_override~false~is_superadmin~false)~sections~(show~(~(dataset~'!mixpanel~value~(name~'ingest~resourceType~'events)~resourceType~'events~profileType~null~search~'~dataGroupId~null~math~'total~perUserAggregation~null~property~null))~group~(~(dataset~'!mixpanel~value~'fileType~resourceType~'events~profileType~null~search~'~dataGroupId~null~propertyType~'string~typeCast~null~unit~null))~time~(~(dateRangeType~'in*20the*20last~unit~'day~window~(unit~'month~value~3))))~legend~(individualSelectedSegments~(selected~(~(key~'Metadata)))))), the most we've seen in the past 3 months is 23, and that was a pretty big outlier.  5-10 is a much more normal range.

I think adding pre-flight and error handling to updating these facets will suffice in terms of reliability & brittleness.